### PR TITLE
Add docs for windows performance analyzer

### PR DESCRIPTION
**Pull request by jasonwilliams**
This adds some new documentation for profiling Boa with Windows Performance Analyzer.
This doesn't replace the measureme/crox effort, but offers another way of profiling Boa which may be more suitable to some users. 

Please try the guide out and let me know if it makes sense or is missing some steps.

[Rendered](https://github.com/jasonwilliams/boa/blob/7903ce981b1dedbbf03d2c9365cdfd437e59da51/docs/profiling_with_wpa.md)

### CODE REVIEW

File: docs/profiling_with_wpa.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+Windows have a performance analyzing tool that creates graphs and data tables of Event Tracing for Windows (ETW) events. You can analyze your whole system but also individual binaries too.
+This can be used for performance tracing with Rust programs like Boa.
+
+For this to work you will need to have your code running in native windows and not WSL.
+
+- First off, install [UI For ETW](https://github.com/google/UIforETW). You can grab the latest version from the releases page.
+- Then install Windows Performance Toolkit, you can grab it from [here](https://learn.microsoft.com/en-us/windows-hardware/get-started/adk-install)
+- [Optional] - Follow [Improved Symbol Setup](#improved-symbol-setup) to improve symbol loading. So that its ready once you open up WPA. This points WPA to your pdb files inside your target directory so you can see the function names instead of just addresses.
+- Open up `UI for ETW`, click on the `Start Tracing` button and then run Boa in dev mode (hopefully a script long enough for it to record something meaningful).
+- When the program has finished click `Save Trace Buffers`, then on your trace below you can right click it and open with WPA
```
Comment (nekevss) [inline docs/profiling_with_wpa.md:12]:
```suggestion
- Open up `UI for ETW`, click on the `Start Tracing` button and then run Boa in dev mode (hopefully a script long enough for it to record something meaningful).
- When the program has finished running, click `Save Trace Buffers` to create the trace.
- Right click on the your newly created trace and open it with WPA
```

File: docs/profiling_with_wpa.md [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+- **CPU Usage (Sampled)** Periodically sampling the CPU usage at predefined intervals. We will be using this view mainly
+- **CPU Usage (Precise)** Offers more detailed and accurate information about CPU usage by using precise event-based tracing. By default Wait Analysis is enabled, which is useful for debugging deadlocks and other issues. However we will not be using this view.
+
+
+Windows Performance Analyzer is analzying everything on your system (which is useful in cases where we may not anticipate outside factors affecting our program).
+
+However we only care about the boa process, so we should filter out everything else. We can do this by right clicking boa.exe and filtering down to just this.
+![filter](./img/filter_to_boa.png)
+
+Once you've done this navigate to the CPU Usage (Sampled) tab underneath and click the first symbol on the far right, it should be called `Display Graph and Table`.
```
Comment (nekevss) [inline docs/profiling_with_wpa.md:29]:
```suggestion
Once completed, navigate to the `CPU Usage (Sampled)` section underneath and click the `Display Graph and Table` header icon, which is the first symbol in the upper right hand corner.
```

File: docs/profiling_with_wpa.md [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+Then on the dropdown (same tab) select `Flame By Process, Stack`. If followed correctly you should have something which looks like this:
```
Comment (nekevss) [inline docs/profiling_with_wpa.md:31]:
```suggestion
Then on `CPU Usage (Sampled)` View Preset dropdown, located just to the right of the header title, select the `Flame By Process, Stack` option. If followed correctly, you should see something similar to the below:
```

File: docs/profiling_with_wpa.md [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+![cpu](./img/cpu_usage.png)
+
+Almost there..
+Symbols are not showing, WPA doesn't scan symbols by default. You need to click Trace at the top and then `Load Symbols`.
```
Comment (nekevss) [inline docs/profiling_with_wpa.md:36]:
```suggestion
Symbols are not showing, WPA doesn't scan symbols by default. To load symbols, click Trace in the application menu at the top and then select `Load Symbols`.
```

**Review by nekevss — COMMENTED** [review]
I was able to follow along fairly well. I still need to play with it a bit more than currently, but it's definitely really cool! Should the improved Symbol Setup step be optional?

Just some suggestions regarding the wording of the doc itself.

**Comment by jedel1043** [comment]
We discussed this in our triage and @raskad @nekevss @HalidOdat will try to run it to see if there are any issues on the example.

**Pull request closed without merging**